src/Stratis.Bitcoin.Tests/Consensus/ConsensusTestExtensions.cs

```diff
@@ -26,5 +26,17 @@ public static ChainedHeader GetPeerTipChainedHeaderByPeerId(this ChainedHeaderTr
         {
             return chainedHeaderTree.GetChainedHeadersByHash()[chainedHeaderTree.GetPeerTipsByPeerId()[peer]];
         }
+
+        public static bool HaveBlockDataAvailabilityStateOf(this ConnectNewHeadersResult connectNewHeadersResult, BlockDataAvailabilityState blockDataAvailabilityState)
+        {
+            if (connectNewHeadersResult.DownloadFrom == null || connectNewHeadersResult.DownloadTo == null) return false;
+            ChainedHeader chainedHeader = connectNewHeadersResult.DownloadTo;
+            while (chainedHeader.Height > connectNewHeadersResult.DownloadFrom.Height)
+            {
+                if (chainedHeader.BlockDataAvailability != blockDataAvailabilityState) return false;
+                chainedHeader = chainedHeader.Previous;
+            }
+            return true;
+        }
```

Comment:
Almost good here, let's just polish it with code style compliance.

That would mean brackets on line 32, empty line after line 32, empty line after line 36 and also after line 38.

Also the wild condition should be greater or equal – imagine download from equal download to.